src/components/app/MenuButtons/Publish.js

```diff
@@ -416,47 +434,89 @@ class DownloadSize extends React.PureComponent<
 }
 
 type DownloadButtonProps = {|
-  +compressedProfileBlobUrlPromise: Promise<string>,
+  // The generation is a number that only increases, and does so each time a profile
+  // sanitization option changes, or a new profile is generated. It should be increased
+  // any time a new sanitized profile is generated. This is how we know to update
+  // this component.
+  +generation: number,
```

Comment:
I think the same remark about `generation` applies here as well.
Setting `compressedProfileObjectUrl: null` can be done in `unwrapPromise` before resolving the promise, and otherwise in `componentDidUpdate` we can check if the promise changed.

And then all the machinery around the generation can be removed.